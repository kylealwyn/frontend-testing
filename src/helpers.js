/*
 * Recursively flatten array
 * @param {Array} source
 * @returns {Array} - Flattened array
 **/
export const flattenArray = (source = []) =>
  source.reduce((arr, node) => Array.isArray(node) ? arr.concat(flattenArray(node)) : arr.concat(node), []);
