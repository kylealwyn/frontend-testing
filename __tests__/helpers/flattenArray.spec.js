import { flattenArray } from '../../src/helpers';

describe('helpers', () => {
  describe('#flattenArray', () => {
    it('should flatten an array of arrays', () => {
      const arrayToFlatten = [[1], [2], [3]]
      expect(flattenArray(arrayToFlatten)).toMatchSnapshot();
    });

    it('should flatten nested arrays', () => {
      const arrayToFlatten = [[1, ['bob']], [{ hello: 'world'}, [4]], [5, [false, [7]]]];
      expect(flattenArray(arrayToFlatten)).toMatchSnapshot();
    });
  });
});
