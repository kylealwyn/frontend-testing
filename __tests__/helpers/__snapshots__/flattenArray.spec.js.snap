// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers #flattenArray should flatten nested arrays 1`] = `
Array [
  1,
  "bob",
  Object {
    "hello": "world",
  },
  4,
  5,
  false,
  7,
]
`;
